### PR TITLE
Add Config::get_bool

### PR DESCRIPTION
**Pull request by lavoiesl**
Coalesce and enforce the value of a config to a boolean

### CODE REVIEW

**Comment by lavoiesl** [comment]
Being done for https://github.com/Shopify/dev/pull/1757

File: lib/cli/kit/config.rb [hunk 1]
```diff
@@ -27,6 +27,18 @@ def get(section, name)
         all_configs.dig("[#{section}]", name) || false
       end
 
+      # Coalesce and enforce the value of a config to a boolean
+      def get_bool(section, name)
+        case get(section, name).to_s
```
Comment (jules2689) [inline lib/cli/kit/config.rb]: should we downcase?

File: test/cli/kit/config_test.rb [hunk 2]
```diff
@@ -25,6 +25,42 @@ def test_config_get_returns_false_for_not_existant_key
...
+
+        @config.set('section', 'foo-key', false)
+        assert_equal('false', @config.get('section', 'foo-key'))
+        assert_equal(false, @config.get_bool('section', 'foo-key'))
+      end
+
+      def test_config_get_bool_on_invalid
+        @config.set('section', 'foo-key', "yes")
+        assert_equal("yes", @config.get('section', 'foo-key'))
+        begin
```
Comment (jules2689) [inline test/cli/kit/config_test.rb]: you can do:
```ruby
e = assert_raises CLI::Kit::Abort do
  @config.get_bool('section', 'foo-key')
end
assert_equal "Invalid config: section.foo-key is expected to be true or false", e.message
```

File: lib/cli/kit/config.rb (hunk 1 above)
Comment (burke) [reply lib/cli/kit/config.rb]: I'd rather it be explicitly `true` or `false` if we're going to start trending toward types.

Comment (lavoiesl) [reply lib/cli/kit/config.rb]: I thought about it, but this enforces consistency in the config files, so I’m ok either way.

Comment (jules2689) [reply lib/cli/kit/config.rb]: Im ok either way as well

**Pull request merged**